### DISCUSSION TITLE
Does @override will be mandatory in the future?

### DISCUSSION THREAD

**Discussion by gcattan** [mobxjs/mobx#3494] [General] [answered]
Hi, 
I understood in a previous discussion that the @override decorator is not mandatory for now.
Will this be always be the case? Or is there plan to make this @override required in the future?
I have to provide a few recommendation to the team, so I would like to clarify whether or not it is a good practice to put it.

**Comment by urugator** [comment, marked as answer]
> required in the future

Will likely depend on ES decorators spec status/implementation.
https://github.com/tc39/proposal-decorators

> it is a good practice to put it

It is:
1) It makes sure the field in super is decorated, which provides similar benefits to TS's `override` keyword:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-3.html#override-and-the---noimplicitoverride-flag
2) It's easier to eventually remove it than add it where necessary.
3) It's possible you will have to replace it with different decorator (again depens on ES spec) - besides being easier to locate, we can change the `override` implementation to provide a friendly warning/instruction (or perhaps even a migration tool).

**Reply by gcattan** [reply]
Ok, super, thanks!
